WCFDataService/Client/System/Data/Services/Client/Materialization/HttpWebResponseMessage.cs

```diff
@@ -47,6 +47,8 @@ public class HttpWebResponseMessage : IODataResponseMessage, IDisposable
         /// <summary>HttpWebResponse instance.</summary>
         private HttpWebResponse httpWebResponse;
 
+        /// <summary>The IODataResponseMessage to be disposed together with this HttpWebResponseMessage.</summary>
+        private IODataResponseMessage underlyingResponseMessage;
```

Comment:
If we use a finally to dispose responseMessage explicitly in the caller of ExecuteQuery, seems we do not need to save the message here.